src/reducers/authReducer.ts

```diff
@@ -0,0 +1,36 @@
+import { createSlice } from '@reduxjs/toolkit';
+import type { PayloadAction } from '@reduxjs/toolkit';
+import { User } from '../utils/types';
+
+interface AuthState {
+  user: User;
+}
+
+interface AuthState {
+  user: User;
+}
+
+const initialState: AuthState = {
+  user: {
+    token: null,
+  },
+};
+const authSlice = createSlice({
+  name: 'counter',
+  initialState,
+  reducers: {
+    login: (state, action: PayloadAction<User>) => ({
+      ...state,
+      user: action.payload,
+    }),
+    logout: (state) => ({
+      ...state,
+      initialState,
+    }),
```

Comment:
@musllim I had suggested a better implementation on this line. 
```suggestion
```

Reply:
Assignment to property of function parameter 'state'.eslint[no-param-reassign](https://eslint.org/docs/rules/no-param-reassign)

Reply:
It looks better but linter can accept it with that error